### PR TITLE
[ODS-4238] Integrate TestSdk Generation

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

File: logistics/scripts/Invoke-SdkGen.ps1 [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+.parameter apiUrl
+    API URL for the Integration Test Harness
+.parameter environmentFilePath
+    Path to environment file used to run Postman integration tests
+
+    .EXAMPLE
+    PS> Invoke-SdkGen.ps1
+#>
+param(
+    [string] $configurationFile = $(Get-RepositoryResolvedPath "logistics\scripts\smokeTestHarnessConfiguration.json"),
```
Comment (jamessmckay) [inline logistics/scripts/Invoke-SdkGen.ps1:22]: we should be using forward slashes for paths to enable powershell core in the future

File: logistics/scripts/Invoke-SdkGen.ps1 [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    [string] $apiUrl = "http://localhost:8765",
+    [string] $environmentFilePath = (Resolve-Path "$PSScriptRoot\modules")
```
Comment (jamessmckay) [inline logistics/scripts/Invoke-SdkGen.ps1:24]: this should be `(Resolve-Path -Path (Join-Path -Path $PSScriptRoot" -ChildPath "modules")).Path`

File: logistics/scripts/Invoke-SdkGen.ps1 [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+)
+
+$ErrorActionPreference = 'Stop'
+
+$error.Clear()
+
+& "$PSScriptRoot\..\..\logistics\scripts\modules\load-path-resolver.ps1"
```
Comment (jamessmckay) [inline logistics/scripts/Invoke-SdkGen.ps1:31]: use forward slashes

Comment (jamessmckay) [reply logistics/scripts/Invoke-SdkGen.ps1:31]: should we be using import module instead?

File: logistics/scripts/modules/TestHarness.psm1 [hunk 4]
```diff
@@ -61,6 +61,18 @@ function Get-TestHarnessExecutable {
     return $testHarnessExecutable
 }
 
+function Invoke-SdkGenConsole {
+    Param(
+        [string] $buildConfiguration = "Debug"
+    )
+    
+    $sdkGenConsoleExecutableFolder = "$(Get-RepositoryResolvedPath "\Utilities\SdkGen\EdFi.SdkGen.Console")\bin\$buildConfiguration\netcoreapp3.1"
```
Comment (jamessmckay) [inline logistics/scripts/modules/TestHarness.psm1:69]: not sure if we should hard code the full location. why not search for the file using a glob pattern?

File: logistics/scripts/modules/TestHarness.psm1 [hunk 5]
```diff
@@ -61,6 +61,18 @@ function Get-TestHarnessExecutable {
...
+    $params = "/c EdFi.SdkGen.Console.exe -m http://localhost:8765/metadata?sdk=true -v 2.4.15 --include-profiles --include-composites --include-identity"
```
Comment (jamessmckay) [inline logistics/scripts/modules/TestHarness.psm1:70]: should this endpoint come from the configuration file?

File: logistics/scripts/Invoke-SdkGen.ps1 [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    API URL for the Integration Test Harness
+.parameter environmentFilePath
+    Path to environment file used to run Postman integration tests
+
+    .EXAMPLE
+    PS> Invoke-SdkGen.ps1
+#>
+param(
+    [string] $configurationFile = $(Get-RepositoryResolvedPath "logistics\scripts\smokeTestHarnessConfiguration.json"),
+    [string] $apiUrl = "http://localhost:8765",
```
Comment (jamessmckay) [inline logistics/scripts/Invoke-SdkGen.ps1:23]: should this endpoint be pulled from the configuration file?

File: logistics/scripts/modules/TestHarness.psm1 [hunk 7]
```diff
@@ -61,6 +61,18 @@ function Get-TestHarnessExecutable {
...
+    Write-Host "$sdkGenConsoleExecutableFolder $params" -ForegroundColor Magenta
+    Start-Process -FilePath cmd -WorkingDirectory $sdkGenConsoleExecutableFolder -verb runas -ArgumentList ($params)
+
```
Comment (jamessmckay) [inline logistics/scripts/modules/TestHarness.psm1:73]: remove blank line

**Review by jamessmckay — CHANGES_REQUESTED** [review]

**Comment by jamessmckay** [comment]
In general we should be using forward slashes for the paths going forward to make the script more compliant with powershell core.

File: logistics/scripts/Invoke-SdkGen.ps1 (hunk 2 above)
Comment (jamessmckay) [reply logistics/scripts/Invoke-SdkGen.ps1:24]: I am confused here you declared the type to be a string, and resolve path returns an object.

File: logistics/scripts/Invoke-SdkGen.ps1 (hunk 1 above)
Comment (semalaiappan) [reply logistics/scripts/Invoke-SdkGen.ps1]: I fixed it . Thanks

File: logistics/scripts/Invoke-SdkGen.ps1 (hunk 2 above)
Comment (semalaiappan) [reply logistics/scripts/Invoke-SdkGen.ps1]: I fixed it . Thanks

File: logistics/scripts/Invoke-SdkGen.ps1 (hunk 3 above)
Comment (semalaiappan) [reply logistics/scripts/Invoke-SdkGen.ps1]: yes I fixed it . Thanks

File: logistics/scripts/modules/TestHarness.psm1 (hunk 7 above)
Comment (semalaiappan) [reply logistics/scripts/modules/TestHarness.psm1]: yes I fixed it . Thanks

File: logistics/scripts/Invoke-SdkGen.ps1 (hunk 6 above)
Comment (semalaiappan) [reply logistics/scripts/Invoke-SdkGen.ps1:23]: No . It's defaultvalue from  in $apiUrl parameter value . Same way it's there in Invoke-PostmanIntegrationTests also .

File: logistics/scripts/modules/TestHarness.psm1 (hunk 5 above)
Comment (semalaiappan) [reply logistics/scripts/modules/TestHarness.psm1]: I updated with $apiUrl here

**Review by jamessmckay — APPROVED** [review]

**Comment by jamessmckay** [comment]
waiting on teamcity changes to finalized before merging

**Comment by semalaiappan** [comment]
image

**Comment by semalaiappan** [comment]
@jamessmckay I added noPackaging  flag to disable Artifact paths settings & avoid package creation for this flow

File: logistics/scripts/modules/TestHarness.psm1 [hunk 8]
```diff
@@ -61,6 +61,20 @@ function Get-TestHarnessExecutable {
...
+        [Parameter(Mandatory = $true)] [string] $apiUrl, 
+        [string] $buildConfiguration = "Debug"
+    )
+    
+    $sdkGenConsoleFolder = (Get-RepositoryResolvedPath "\Utilities\SdkGen\EdFi.SdkGen.Console")
+    $sdkGenConsoleExeName = (Get-Item $sdkGenConsoleFolder).Name
+    $sdkGenConsoleExecutableFolder = "$sdkGenConsoleFolder\bin\$buildConfiguration\netcoreapp3.1"
+    $params = "/c $sdkGenConsoleExeName.exe -m "+ $apiUrl+"/metadata?sdk=true -v 2.4.15 --include-profiles --include-composites --include-identity"
+    Write-Host "$sdkGenConsoleExecutableFolder $params" -ForegroundColor Magenta
+    Start-Process -FilePath cmd -WorkingDirectory $sdkGenConsoleExecutableFolder -verb runas -ArgumentList ($params) -PassThru -Wait
```
Comment (jamessmckay) [inline logistics/scripts/modules/TestHarness.psm1:75]: we should be using & $params to run this build.

**Review by jamessmckay — CHANGES_REQUESTED** [review]

File: logistics/scripts/modules/TestHarness.psm1 (hunk 8 above)
Comment (semalaiappan) [reply logistics/scripts/modules/TestHarness.psm1]: https://intedfitools1.msdf.org/buildConfiguration/OdsPlatform_OdsApi_Packages_EdFiOdsApiTestSdk/128693

**Comment by semalaiappan** [comment]
@jamessmckay  The alternative solution is provided as below

**Review by jamessmckay — APPROVED** [review]

**Pull request merged**